WebView/app/src/androidTest/java/com/android/samples/webviewdemo/MainActivityTest.kt

```diff
@@ -0,0 +1,158 @@
+import androidx.test.espresso.web.webdriver.DriverAtoms.findElement
+import androidx.test.espresso.web.webdriver.DriverAtoms.getText
+import androidx.test.espresso.web.webdriver.DriverAtoms.webClick
+import androidx.test.espresso.web.webdriver.Locator
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.ActivityTestRule
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertTrue
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.CoreMatchers.containsString
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+
+/**
+ * Launch, interact, and verify conditions in an activity that has a WebView instance.
+ */
+@RunWith(AndroidJUnit4::class)
+
+
+class MainActivityTest {
+
+    val context = ApplicationProvider.getApplicationContext<Context>()
+
+    @Rule @JvmField
+    val mainActivityRule = ActivityTestRule(MainActivity::class.java)
+    fun afterActivityLaunched() {
+      // Technically we do not need to do this - MainActivity has javascript turned on.
+      // Other WebViews in your app may have javascript turned off, however since the only way
+      // to automate WebViews is through javascript, it must be enabled.
+        onWebView().forceJavascriptEnabled()
+    }
+
+    // Test to check that the drop down menu behaves as expected
+    @Test
+    fun dropDownMenu_SanFran() {
+        mainActivityRule.getActivity()
+        onWebView()
+            .withElement(findElement(Locator.ID, "location"))
+            .perform(webClick()) // Similar to perform(click())
+            .withElement(findElement(Locator.ID, "SF"))
+            .perform(webClick()) // Similar to perform(click())
+            .withElement(findElement(Locator.ID, "title"))
+            .check(webMatches(getText(), containsString("San Francisco")))
+    }
+
+    // Test for checking createJsObject
+    @Test
+    fun jsObjectIsInjectedAndContainsPostMessage() {
+    mainActivityRule.getActivity()
+    onWebView()
+        .check(
+            webMatches(
+                script("return jsObject && jsObject.postMessage != null;", castOrDie(Boolean::class.javaObjectType)),
+                `is`(true)
+            )
+        )
+    }
+
+    @Test
+    fun valueInCallback_compareValueInput_returnsTrue(){
+        mainActivityRule.getActivity()
+        // Setup
+        val jsObjName = "jsObject"
+        val allowedOriginRules = setOf<String>("https://example.com")
+        val message = "hello"
+        // Get a handler that can be used to post to the main thread
+        val mainHandler = Handler (Looper.getMainLooper());
+        val myRunnable = Runnable() {
+            val webView = WebView(context)
+            // Create JsObject
+            createJsObject(
+                webView,
+                jsObjName,
+                allowedOriginRules
+            ) { message -> //save message; call .set()
+            }
+            run() {
```

Comment:
All this should be inside the runnable's "run()" method.

---

nit: there should be a way to rewrite this to be more like a lambda style:

```java
// This is how it would look in Java
mainHandler.post(() -> {
  WebView webView = WebView(context);
  createJsObject(webView, jsObjName, allowedOriginRules, () -> {
    // ...
  });
  // ...
});

Reply:
Ok, I will move that and make the change to a lambda style.